src/dex/bebop/bebop.ts

```diff
@@ -0,0 +1,744 @@
+    limit: number,
+  ): Promise<PoolLiquidity[]> {
+    const prices = await this.getCachedPrices();
+
+    if (!prices) {
+      return [];
+    }
+
+    const pools: PoolLiquidity[] = [];
+
+    for (const [pair, pairData] of Object.entries(prices)) {
+      let liquidityUSD = 0;
+      let token;
+      const [base, quote] = pair.split('/');
+      if (base.toLowerCase() == tokenAddress.toLowerCase()) {
+        const liquidityInQuote = this.getMaxLiquidity(pairData.bids);
+        token = {
+          address: quote,
+          decimals: this.tokensMap[quote.toLowerCase()].decimals,
+        };
+        const quoteTokenUsd = await this.dexHelper.getTokenUSDPrice(
+          token,
+          BigInt(Math.round(liquidityInQuote)),
+        );
+        liquidityUSD = liquidityInQuote * quoteTokenUsd;
+      } else if (quote.toLowerCase() == tokenAddress.toLowerCase()) {
+        const liquidityInBase = this.getMaxLiquidity(pairData.asks);
+        token = {
+          address: base,
+          decimals: this.tokensMap[base.toLowerCase()].decimals,
+        };
+        const baseTokenUsd = await this.dexHelper.getTokenUSDPrice(
+          token,
+          BigInt(Math.round(liquidityInBase)),
+        );
+        liquidityUSD = liquidityInBase * baseTokenUsd;
+      }
+      if (liquidityUSD) {
+        if (pools.length === 0) {
+          pools.push({
+            exchange: this.dexKey,
+            address: this.settlementAddress,
+            connectorTokens: [
+              {
+                address: quote,
+                decimals: this.tokensMap[quote.toLowerCase()].decimals,
+                symbol: this.tokensMap[quote.toLowerCase()].ticker,
+              },
+            ],
+            liquidityUSD,
+          });
+        } else {
+          pools[0].liquidityUSD += liquidityUSD;
+          pools[0].connectorTokens.push({
+            address: quote,
+            decimals: this.tokensMap[quote.toLowerCase()].decimals,
+            symbol: this.tokensMap[quote.toLowerCase()].ticker,
+          });
+        }
+      }
+    }
+
+    return pools
+      .sort((a, b) => b.liquidityUSD - a.liquidityUSD)
+      .slice(0, limit);
+  }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: BebopData,
+    side: SwapSide,
+  ): DexExchangeParam {
+    const { tx } = data;
+
+    assert(tx !== undefined, `${this.dexKey}-${this.network}: tx undefined`);
+
+    return {
+      exchangeData: tx.data,
+      needWrapNative: this.needWrapNative,
+      dexFuncHasRecipient: false,
+      targetExchange: this.settlementAddress,
+      returnAmountPos: undefined,
+    };
+  }
+
+  async preProcessTransaction(
+    optimalSwapExchange: OptimalSwapExchange<BebopData>,
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    options: PreprocessTransactionOptions,
+  ): Promise<[OptimalSwapExchange<BebopData>, ExchangeTxInfo]> {
```

Comment:
@JackieJoo I don't see pools restriction logic in preProcessTransaction. Usually we have this logic for RFQs